src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -48,4 +51,21 @@ public String getBucket() {
     return bucket;
   }
 
+  @Override
+  public void validate() {
+    super.validate();
+    if (!containsMacro("bucket") && bucket != null) {
+      String bucketName = BigQueryUtil.getBucketNameWithoutScheme(bucket);
```

Comment:
it would be better if the getBucket() methods remove the scheme and also return null if the bucket is empty. Then you just use getBucket() everywhere instead of forcing every caller to check for empty and remove the scheme.